genie-app/build.gradle

```diff
@@ -45,3 +41,70 @@ bootRun {
     addResources = false
     systemProperty "spring.profiles.active", env
 }
+
+def dockerDir = new File(project.buildDir, "/docker")
+
+def isLatest() {
+    return (boolean) !project.version.toString().contains("SNAPSHOT") && !project.version.toString().contains("-rc.")
+}
+
+def getTags(String appName) {
+    def tags = ["netflixoss/${appName}:${project.version}"]
+    if (isLatest()) {
+        tags.add("netflixoss/${appName}:latest")
+    }
+    return tags
+}
+
+task dockerCreateInputDir(type: Copy, group: "Docker") {
+    dependsOn tasks.jar, tasks.bootRepackage
+    from tasks.jar
+    from new File(project.projectDir, "src/main/docker/Dockerfile")
+    into dockerDir
+}
+
+task dockerBuildImage(type: Exec, group: "Docker") {
+    dependsOn tasks.dockerCreateInputDir
+    workingDir dockerDir
+
+    def appName = jar.baseName.toString()
+
+    def commandArgs = new ArrayList<>()
+    commandArgs.add("docker")
+    commandArgs.add("build")
+    commandArgs.add("--build-arg")
+    commandArgs.add("JAR_NAME=${appName}")
+    commandArgs.add("--build-arg")
+    commandArgs.add("VERSION=${project.version}")
+    for (String tag : getTags(appName)) {
+        commandArgs.add("-t")
+        commandArgs.add(tag)
+    }
+    commandArgs.add(".")
```

Comment:
You may want to consider using ```--label``` for adding some additional metadata related to this build, such as the release name in case only ```latest``` is the only available tag.